pkg/aggregator/sender_test.go

```diff
@@ -666,3 +666,25 @@ func TestCheckSenderHostname(t *testing.T) {
 		})
 	}
 }
+
+func TestSenderGaugeWithTimestampValidation(t *testing.T) {
+	// this test not using anything global
+	// -
+
+	s := initSender(checkID1, "")
+
+	s.sender.GaugeWithTimestamp("my.gauge_with_timestamp", 42, "my-hostname", nil, time.Now())
+	metricSample := (<-s.itemChan).(*senderMetricSample)
+	assert.EqualValues(t, 42, metricSample.metricSample.Value)
```

Comment:
Should we assert the timestamp too here?

Reply:
Sure, added in 348fbeb thanks!